libminifi/include/core/ConfigurableComponent.h

```diff
@@ -153,14 +153,14 @@ class ConfigurableComponent {
    *
    * @return vector of property keys
    */
-  std::vector<std::string> getDynamicPropertyKeys();
+  std::vector<std::string> getDynamicPropertyKeys() const;
 
   /**
    * Returns a vector all properties
    *
    * @return map of property keys to Property instances.
    */
-  std::map<std::string, Property> getProperties();
+  std::map<std::string, Property> getProperties() const;
```

Comment:
Did you not change to const ref map because current implementations expect to modify this map?

Reply:
Nope, the map is created upon execution, (a merge of dynamic and static properties), so there is no real point in returning const ref.